core/src/main/scala/kafka/controller/ControllerState.scala

```diff
@@ -120,8 +120,13 @@ object ControllerState {
     def value = 19
   }
 
+  case object UpdateMetadataResponseReceived extends ControllerState {
+    def value = 20
+  }
+
   val values: Seq[ControllerState] = Seq(Idle, ControllerChange, BrokerChange, TopicChange, TopicDeletion,
     AlterPartitionReassignment, AutoLeaderBalance, ManualLeaderBalance, ControlledShutdown, IsrChange, LeaderAndIsrResponseReceived,
     LogDirChange, ControllerShutdown, UncleanLeaderElectionEnable, TopicUncleanLeaderElectionEnable, ListPartitionReassignment,
-    TopicDeletionFlagChange, PreferredControllerChange, TopicMinInSyncReplicasConfigChange, SkipControlledShutdownSafetyCheck)
+    TopicDeletionFlagChange, PreferredControllerChange, TopicMinInSyncReplicasConfigChange, SkipControlledShutdownSafetyCheck,
```

Comment:
Please move the UpdateMetadataResponseReceived before TopicDeletionFlagChange, so that the ones at the end are from LI hotfixes.